File: model/src/main/java/com/grocerylist/model/Product.java
package com.grocerylist.model;

import com.grocerylist.constants.Unit;
import lombok.Data;

import javax.persistence.*;
/**
 * @author  Adam Mendak
 * @version 1.0
 * @since   2019-10-20
 *
 * Abstract class which contains fields for holding entered product data
 */
@Data
@Entity
@Table(name = "PRODUCT")
public class Product extends AbstractTimestampableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "NAME", nullable = false)
    private String name;

    @Column(name = "UNIT")
    @Enumerated(EnumType.STRING)
    private Unit unit;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.DETACH)
    @JoinColumn(name = "PRODUCT_CATEGORY_ID")
    private ProductCategory productCategory;

}

File: frontend/groceryFront/src/app/dish/product.ts
export interface IProduct {
  name: string;
  unit: string;
  productCategory: string;
}

export class Product implements  IProduct {
  name: string;
  productCategory: string;
  unit: string;

  setName(value: string) {
    this.name = value;
  }

  setProductCategory(value: string) {
    this.productCategory = value;
  }

  setUnit(value: string) {
    this.unit = value;
  }
}
